server/util/repeatingRide.ts

```diff
@@ -0,0 +1,115 @@
+import { v4 as uuid } from 'uuid';
+import moment from 'moment';
+import { Condition } from 'dynamoose';
+import schedule from 'node-schedule';
+import { AnyDocument } from 'dynamoose/dist/Document';
+import { Ride, RideType } from '../models/ride';
+
+export default function initSchedule() {
+// run function at 10:05:00am every day
+  schedule.scheduleJob('0 5 10 * * *', () => {
+    createRepeatingRides();
+  });
+}
+
+
+function createRepeatingRides() {
+  let tomorrowDate = moment(new Date());
+  tomorrowDate = tomorrowDate.add(1, 'days');
+  const tomorrowDateString = tomorrowDate.format('MM/DD/YYYY');
+
+  const tomorrowDay = tomorrowDate.weekday();
+
+  // condition: recurring ride that includes tomorrowDay
+  const condition = new Condition()
+    .where('recurring')
+    .eq(true)
+    .and()
+    .where('recurringDays')
+    .contains(tomorrowDay);
+
+  Ride.scan(condition).exec((_, data) => {
+    data?.forEach(async (masterRide) => {
+      const {
+        id, rider, driver, startLocation, endLocation, startTime, endTime, endDate, edits,
+      } = masterRide.toJSON() as RideType;
+
+      // only continue if the endDate has not passed
+      if (endDate && endDate >= tomorrowDateString) {
```

Comment:
endDate should be formatted into YYYY-MM-DD too (or changed to be stored as such), or else some comparisons will be incorrect (ex. 01/01/2021 < 01/02/2020)

Reply:
just formatted it to YYYY-MM-DD